src/clouds.h

```diff
@@ -0,0 +1,45 @@
+#include <vector>
+// extern "C" {
```

Comment:
I think these can be taken out. In fact, it may be good to go through all of the header/source files you've changed to double check that you're not over-including stuff. Fogleman seems to include the include statements in the source files, with no mention of them in header files.